### PR TITLE
Operation block list from gravity

### PR DESCRIPTION
**Pull request by PromoFaux**
Fixes/address #247

This is mainly for the advanced users. 

`Gravity.sh` now checks for the existence of `/etc/pihole/adlists.list`. 
If it finds it, then it will load it's list of adlists from the urls contained therein (one per line). 
If not, it will get the domains from `/etc/pihole/adlists.default`

### CODE REVIEW

**Comment by PromoFaux** [comment]
I've tested this myself, but if I could get confirmation from or @dschaper or @Mcat12  before I merge, that would be great.

**Comment by Mcat12** [comment]
Is the adlist.default being copied in the installer? I'm getting:
```
:::
::: You are root.
::: Neutrino emissions detected...
:::
::: No custom adlist file detected, reading from default file.../usr/local/bin/gravity.sh: line 92: /etc/pihole/adlists.default: No such file or directory
 done!                                                                                                                                                                                                                                      .
:::
:::
::: Aggregating list of domains.. done!
::: Formatting list of domains to remove comments... done!
::: 0 domains being pulled in by gravity...
::: Removing duplicate domains... done!
::: 0 unique domains trapped in the event horizon.
::: Formatting domains into a HOSTS file...
:::
::: Adding 0 ad list sources to the whitelist... done!
::: Running whitelist script to update HOSTS file... 45 domains whitelisted!
::: Running blacklist script to update HOSTS file... 9 domains blacklisted!
:::
::: Refresh lists in dnsmasq.. done!
```

**Comment by PromoFaux** [comment]
Yep, as in here ef99fe2

**Comment by Mcat12** [comment]
So it copies the adlist when it detects a previous version, but does it copy on a clean install?

**Comment by PromoFaux** [comment]
OK. so...

Both a fresh install and an upgrade will copy `adlists.default` to `/etc/pihole/adlists.default`.

This file is not for editing, but rather a backup of the default adlists.

If a user wishes to customise the list, then they make a copy of that list and call it `adlists.list`, then edit it to their heart's content.

Install script is completely unaware of the existence of `adlists.list` it only cares that `adlist.default` is there. `gravity.sh` will check for the existence of `adlists.list` and if it exists, use that instead of `adlist.default`

**Comment by Mcat12** [comment]
What line copies the default list when it is a fresh install? I only see the line when it's an existing install.

**Comment by PromoFaux** [comment]
`$SUDO cp -o /etc/.pihole/adlists.default /etc/pihole/adlists.default`

note the `.` in the first path. It's the git directory the installer clones into

**Comment by PromoFaux** [comment]
Yeah I was being blind. Other than that, are you happy with the update?
I will chuck this in at the same time as the install script hotfixes, if so.

**Comment by PromoFaux** [comment]
@Mcat12 check again please :)

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -431,9 +431,9 @@ installCron(){
 
 runGravity(){
 	$SUDO echo ":::"
-	$SUDO echo "::: Preparing to run gravity.sh to refresh hosts..."
+	$SUDO echo "::: Preparing to run gravity.sh to refresh hosts..."	
 	if ls /etc/pihole/list* 1> /dev/null 2>&1; then
-		echo "::: Cleaning up previous install (preserving whitelist/blacklist)"
+		echo "::: Cleaning up previous install (preserving whitelist/blacklist)"
```
Comment (Mcat12) [inline automated install/basic-install.sh]: There's some random whitespace here.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -431,9 +431,9 @@ installCron(){
 
 runGravity(){
 	$SUDO echo ":::"
-	$SUDO echo "::: Preparing to run gravity.sh to refresh hosts..."
+	$SUDO echo "::: Preparing to run gravity.sh to refresh hosts..."
```
Comment (Mcat12) [inline automated install/basic-install.sh]: What's different here?

Comment (DonLexos) [inline automated install/basic-install.sh]: Maybe a different linebreak?

**Comment by Mcat12** [comment]
I'm getting
```
::: Running gravity.sh
:::
:::
::: You are root.
cp: invalid option -- 'o'
Try 'cp --help' for more information.
::: Neutrino emissions detected...
:::
::: No custom adlist file detected, reading from default file.../usr/local/bin/gravity.sh: line 92: /etc/pihole/adlists.default: No such file or directory
 done!                                                                                                                                                      .
:::
:::
::: Aggregating list of domains.. done!
::: Formatting list of domains to remove comments... done!
::: 0 domains being pulled in by gravity...
::: Removing duplicate domains... done!
::: 0 unique domains trapped in the event horizon.
::: Formatting domains into a HOSTS file...
:::
::: Adding 0 ad list sources to the whitelist... done!
::: Running whitelist script to update HOSTS file... 45 domains whitelisted!
::: Running blacklist script to update HOSTS file... 9 domains blacklisted!
:::
::: Refresh lists in dnsmasq.. done!
```

**Comment by Mcat12** [comment]
By taking out that -o option, it works.

**Comment by PromoFaux** [comment]
-o removed.

**Comment by Mcat12** [comment]
This should be good then.

**Pull request merged**